app/views/comments/destroy.js.erb

```diff
@@ -0,0 +1,5 @@
+commentDiv = $("#comments-<%= reference.id %>").find("#comment-<%= @comment.id %>");
+if (commentDiv.siblings().length == 0) { commentDiv.parent(".reply").remove() };
+commentDiv.remove();
+
+$("li#comments-tab-<%= reference.id %>").html("<%= escape_javascript(render 'references/comments_tab', reference: reference) %>");
```

Comment:
Would you mind just decrement the counter instead of re-rendering the entire comments tab? You'd probably want to use JavaScript `parseInt`